plugins/secom_cl.py

```diff
@@ -196,322 +221,463 @@ def sem_roi_to_ccd(escan, ccd, roi, margin=0):
     logging.info("ROI with margin defined at ({:.3e}, {:.3e}, {:.3e}, {:.3e}) m".format(*phys_rect))
 
     # convert physical position to CCD
-    ccd_roi = convert_roi_phys_to_ccd(ccd, phys_rect)
+    ccd_roi = convert_roi_phys_to_ccd(detector, phys_rect)
     if ccd_roi is None:
         logging.error("Failed to find the ROI on the CCD, will use the whole CCD")
-        ccd_roi = (0, 0) + ccd.resolution.value
+        ccd_roi = (0, 0) + detector.resolution.value
     else:
         logging.info("Will use the CCD ROI %s", ccd_roi)
 
     return ccd_roi
 
 
-class GridAcquirer(object):
+class SECOMCLSettingsStream(acqstream.CCDSettingsStream):
+    """
+    A cl settings stream, for a set of points (on the SEM).
+    The live view is just the raw CCD image.
+    """
+    def __init__(self, name, detector, dataflow, emitter, **kwargs):
+        if "acq_type" not in kwargs:
+            kwargs["acq_type"] = model.MD_AT_CL
+
+        super(SECOMCLSettingsStream, self).__init__(name, detector, dataflow, emitter, **kwargs)
```

Comment:
If I understand correctly, you cannot override the original methods because the `__init__` would fail?
If so, don't call the `__init__` of RepetitionStream, and instead, just copy all the code (needed). In practice, this is going to be barely more than you have now. Note, you still need to call the init of the LiveStream, and you can do it with `LiveStream.__init__(self, ....)`